### ISSUE TITLE
How to get the result of our own ffmpeg commands ?

### ISSUE THREAD

**Issue by vinodvinu** [tomaszzmuda/Xabe.FFmpeg#302]
Hi,
The documentation says that we can use our ffmpeg commands in xabe.ffmpeg. But how do we get the result of that command ? Suppose i want use a command to detect the black frames using "blackdetect" feature. When using this command ffmpeg will spit up the time of the black frame's start and end. I need to get this time.

**Comment by tomaszzmuda** [comment]
Hello,

The only possible solution is to parse output text from ffmpeg. You will get this text in OnDataReceived event.

**Comment by vinodvinu** [comment]
Thanks for the reply. I tested it and it worked. Just experimented with "-progress pipe:1" switch. But results are receiving different events. I mean, it's depend upon what command we use in ffmpeg. If I use black detection command, the detected frame's start & end times are coming through the Process.ErrorDataReceived event. But if i use the command to find the total frame number, then the result is coming through the Process.OutputDataReceived event. At first, I just got confused a little with this behavior. But the interesting part is that, i don't need xabe.ffmpeg anymore in this scenario. i can generate events with "-progress" switch. All i need is to write different handlers for each command. I wrapped up the process & startinfo code in a sub and this sub is receiving a custom command and two event handlers  as parameters.
